app/webpacker/components/ScrambleMatcher/index.jsx

```diff
@@ -61,6 +61,22 @@ function ScrambleMatcher({ wcifEvents, competitionId, initialScrambleFiles }) {
     mutationFn: () => submitMatchedScrambles(competitionId, matchState),
   });
 
+  const roundIds = useMemo(() => _.flatMap(wcifEvents, 'rounds').map((r) => r.id), [wcifEvents]);
+
+  const missingIds = useMemo(() => {
+    if (!matchState) return [];
```

Comment:
When does this happen? The initializer for the reducer uses an actual `reduce` (lol) with an empty `{}` as initial argument. Even that has a "valid" (although empty, hence the quotes) set of keys.

Reply:
it doesn't, but it would give an error that scrambles are missing for all events otherwise, do we want that functionality?

Reply:
Mh, it is an efficient way of signaling to the user that "nothing has been uploaded yet".
Maybe if you rephrase the wording a little bit, or do a simple `if missingRoundIds.length === roundIds.length then display kind "please upload something" else display actual error` thingy.

But to be honest, in terms of code lines that would be at least as complicated as the "if empty return empty" special case you're handling right now. So I'm not very firm on this and leave it up to you. Just _some_ way of "hey there is no data yet why don't you upload something" would be nice

Reply:
actually, I do like just showing the user that it's all missing scramble sets when you first open it